### ISSUE TITLE
On 'Phone or email' page, 'log in' click from footer does not focus on 'Phone or email' field

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#18560]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app on android or android chrome
2. If logged in then log out
3. Scroll to bottom and click on 'Log in'
4. Observe that app does not focus on 'Phone or email' field
5. Enter email or phone number and click on continue
6. Scroll to bottom and click on 'Log in' and observe that app now focuses on 'Phone or email' field

## Expected Result: 
App should focus on 'Phone or email' field on 'log in' click from footer as it does on magic code page
## Actual Result: 
App does not focus on 'Phone or email' field on 'Phone or email' page when we click on 'log in' from footer

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.11.2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/236697785-4752b75e-4b00-4bec-a9cf-b2b33112f9a6.mp4

https://user-images.githubusercontent.com/43996225/236697789-c40eb9e4-67a8-4a7e-b2b1-251713e93689.mp4

https://user-images.githubusercontent.com/43996225/236697795-a72a729d-e621-40e7-9175-0842e3b2a448.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1683389488743559

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by dukenv0307** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"Phone or email" field is not focused when clicking the buttons in the footer that scroll to the header
### What is the root cause of that problem?
On 'Phone or email' page when we click on  'log in' button, `navigateHome` is called and it has no way to focus the input field and `isVisible` is true now so when `componentDidUpdate` in `LoginForm` is called it will not autofocus again to input field.
On 'Validate code' page, when we click on that button, the page goes back to 'Phone or email' page `isVisible` props is changed from `false` to `true` so that `componentDidUpdate` run the case `this.input.focus`.
https://github.com/Expensify/App/blob/ab9171f953603bb8e4e061f5582a4b7cab3d4a43/src/pages/signin/LoginForm.js#L90-L98 
### What changes do you think we should make in order to solve the problem?
We could create an `inputRef` in `SignInPageLayout` and assign it to `ref` prop of input in `LoginForm`.  We also pass it into `Footer` and  in `navigateHome`  function we would focus the input if `current.route` is the home route.
### What alternative solutions did you explore? (Optional)
We could create a manage focus same as `ReportActionComposeFocusManager`, but I think it is not necessary because we only use it in `SignInPage`.

### Result

[Screencast from 08-05-2023 15:40:47.webm](https://user-images.githubusercontent.com/129500732/236778352-8029faad-72b9-4fff-8e99-be7673332884.webm)

**Comment by puneetlath** [comment]
Yes, I think that's fine though. I don't think the user will expect it to be in that scenario.
